src/webgpu/api/validation/encoding/queries/begin_end.spec.ts

```diff
@@ -9,15 +9,50 @@ TODO:
     - begin 2, end 2
     - }
     - x= {
-        - render pass + occlusion
         - render pass + pipeline statistics
         - compute pass + pipeline statistics
         - }
-- nesting: test whether it's allowed to nest various types of queries
-  (including writeTimestamp inside begin/endable queries).
 `;
 
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
 import { ValidationTest } from '../../validation_test.js';
 
 export const g = makeTestGroup(ValidationTest);
+
+g.test('occlusion_query,begin_end')
+  .desc(
+    `
+Tests that begin/end occlusion query on render pass:
+- begin 0
+- begin 0, end (control case)
+- begin 0, begin 1, end
+- begin 0, end, end
+- end
+  `
+  )
```

Comment:
Consider sketching out the `.cases`/`.subcases` of this test as part of this test plan. That can replace the list in the description and would help understand how the test is intended to work.

Perhaps something like
```ts
[
  { _valid: false, calls: [0] },
  { _valid: true, calls: [0, 'end'] },
  { _valid: false, calls: [0, 1, 'end'] },
  { _valid: false, calls: [0, 'end', 'end'] },
  { _valid: false, calls: ['end'] },
] as const
```